testsuite/features/support/cobbler_test.rb

```diff
@@ -67,30 +81,46 @@ def profile_create(name, distro, location)
     profile_id
   end
 
-  def profile_exists(name)
-    exists('profiles', 'name', name)
+  # every system needs at least a name and a profile
+  def system_create(name, profile)
+    begin
+      system_id = @server.call('new_system', @token)
+    rescue
+      raise 'creating system failed.' + $ERROR_INFO.to_s
+    end
+    begin
+      @server.call('modify_system', system_id, 'name', name, @token)
+      @server.call('modify_system', system_id, 'profile', profile, @token)
+    rescue
+      raise 'modifying system failed.' + $ERROR_INFO.to_s
+    end
+    begin
+      @server.call('save_system', system_id, @token)
+    rescue
+      raise 'saving system failed.' + $ERROR_INFO.to_s
+    end
+    system_id
   end
 
-  def system_exists(name)
-    exists('systems', 'name', name)
+  def system_remove(name)
+    raise "system cannot be found. #{$ERROR_INFO}" unless system_exists(name)
+    begin
+      @server.call('remove_system', name, @token)
+    rescue
+      raise "deleting system failed. #{$ERROR_INFO}"
+    end
   end
 
   def distro_exists(name)
     exists('distros', 'name', name)
   end
 
-  def distro_create(name, kernel, initrd, breed = 'suse')
-    begin
-      distro_id = @server.call('new_distro', @token)
-      @server.call('modify_distro', distro_id, 'name', name, @token)
-      @server.call('modify_distro', distro_id, 'kernel', kernel, @token)
-      @server.call('modify_distro', distro_id, 'initrd', initrd, @token)
-      @server.call('modify_distro', distro_id, 'breed', breed, @token)
-      @server.call('save_distro', distro_id, @token)
-    rescue
-      raise 'creating distribution failed.' + $ERROR_INFO.to_s
-    end
-    distro_id
+  def profile_exists(name)
+    exists('profiles', 'name', name)
+  end
+
+  def system_exists(name)
+    exists('systems', 'name', name)
```

Comment:
could this not be a single function, getting an argument of distro, profile or system? it seems to be the only thing different between them :thinking:

Reply:
Of course we can do this :wink:  As mentioned earlier to Oscar, I just extended the code and stuck to the style already present. I will mention this is the issue I create for improving/refactoring the `CobblerTest` class.

Reply:
https://github.com/SUSE/spacewalk/issues/17382